src/rgw/rgw_rest_swift.cc

```diff
@@ -1491,7 +1491,10 @@ void RGWInfo_ObjStore_SWIFT::list_swift_data(Formatter& formatter,
 
   string ceph_version(CEPH_GIT_NICE_VER);
   formatter.dump_string("version", ceph_version);
-  formatter.dump_int("max_meta_name_length", 81);
+
+  const size_t meta_name_limit = g_conf->osd_max_attr_name_len
+    - strlen(RGW_ATTR_PREFIX);
```

Comment:
FIXME: should be `RGW_AMZ_META_PREFIX`. `RGW_ATTR_PREFIX` is only a substring of it.